### DISCUSSION TITLE
Species not found in gene family

### DISCUSSION THREAD

**Discussion by arunsethuraman** [hahnlab/CAFE5#78] [Q&A] [answered]
Hi folks,
I'm trying to run CAFE5 on a dataset that I formatted based on the output of Orthofinder (attached - hymenoptera_orthogroups.txt), and the binary, rooted, ultrametric tree (attached - hymtree_ultrametric.txt). I keep ending up with a weird error upon running this command:
cafe5 -i hymenoptera_orthogroups.txt -t hymtree_ultrametric.txt
Command line: cafe5 -i hymenoptera_orthogroups.txt -t hymtree_ultrametric.txt
Tric was not found in gene family x

Any thoughts or advice would be greatly appreciated!
Thanks!
Arun
[hymenoptera_orthogroups.txt](https://github.com/hahnlab/CAFE5/files/6193102/hymenoptera_orthogroups.txt)
[hymtree_ultrametric.txt](https://github.com/hahnlab/CAFE5/files/6193110/hymtree_ultrametric.txt)

**Comment by benfulton** [comment, marked as answer]
This generally indicates that your file has Windows-style line endings. You can use the Linux program dos2unix to fix the line endings.

**Reply by sanvi1804** [reply]
Hi
I have the same issue and i tried converting my file from dos2unix also but it didn't work. I am attaching files.
[SpeciesTree_rooted.txt](https://github.com/user-attachments/files/16456514/SpeciesTree_rooted.txt)
[Gene_families.txt](https://github.com/user-attachments/files/16456500/Gene_families.txt)

**Reply by benfulton** [reply]
When I run your files I get the error message, "China was not found in gene family OG0000000". In your tree file you have "Nelumbo nucifera China" (with spaces) and in the families you have "Nelumbo_nucifera_China" (with underscores). Try replacing the spaces with underscores in your tree.

**Comment by arunsethuraman** [comment]
Goodness - such a rookie error on my part! Thank you! This fixed the issue instantly.

**Comment by nahumemss** [comment]
Hi
I have the same issue, but the dos2unix doesn't work and the same message appear to me.

=(

**Comment by benfulton** [comment]
Can you share your tree and family files?

**Reply by Narutoo-s** [reply]
Hi
I have the same issue, but the dos2unix doesn't work and the same message appear to me. Here are my gene _family.txt and tree.txt. would you help me to solve the problem? Thank you!
[gene_family_filter.txt](https://github.com/hahnlab/CAFE5/files/14927492/gene_family_filter.txt)
[tree.txt](https://github.com/hahnlab/CAFE5/files/14927493/tree.txt)

**Reply by benfulton** [reply]
I do not get the "X was not found in gene family Y" error with your files.

**Comment by nahumemss** [comment]
I changed the gene files for .csv and works

**Comment by kat-king** [comment]
Hi, I'm having this same issue but the two solutions above don't work for me. Any suggestions?

**Comment by benfulton** [comment]
Can you share your tree and family files?

**Comment by kat-king** [comment]
Yes :)

[OF_newick_tree.txt](https://github.com/hahnlab/CAFE5/files/9230052/OF_newick_tree.txt)
[transporter_families.txt](https://github.com/hahnlab/CAFE5/files/9230053/transporter_families.txt)

**Comment by benfulton** [comment]
dos2unix should fix this for you. When I open the file in Notepad++ and view symbols, I see the image below, which shows the DOS CR/LF ending. Run dos2unix on the file and it will remove the CR.

![image](https://user-images.githubusercontent.com/46128/183151229-0db04f23-8ae7-4a4f-a19b-990ccf31c11b.png)

**Reply by kat-king** [reply]
Hi, sorry there were some IT issues but it's back up and running now and it worked :) Thank you!!

**Comment by zeydur** [comment]
Hello I have the same issue, I have tried all the of suggestions but still getting the same error. What should I do?

**Comment by benfulton** [comment]
Can you share your tree and family files?

**Comment by timeforthemoonnight** [comment]
Hi, i am having the same problem and have tried all solutions. here are the files

[hog_gene_counts.txt](https://github.com/hahnlab/CAFE5/files/14855507/hog_gene_counts.txt)
[SpeciesTree_rooted_ultra.txt](https://github.com/hahnlab/CAFE5/files/14855508/SpeciesTree_rooted_ultra.txt)

**Reply by benfulton** [reply]
The family file needs to be tab-separated, not space-separated.

**Comment by MaevaTecher** [comment]
Hi Ben, thank you for all the suggestions. I have also been trying on my side dos2unix, formatting and csv but I still have the issue. I hope you could look at my files and tell me what is wrong with them. I have the same message about species X not found in Gene family Y. Thanks in advance :)
[species_tree_mt.txt](https://github.com/hahnlab/CAFE5/files/15317731/species_tree_mt.txt)
[orthogroups_mt.txt](https://github.com/hahnlab/CAFE5/files/15317733/orthogroups_mt.txt)

**Reply by benfulton** [reply]
Your orthogroups file needs a description field as the first field. If you look at the sample file file it looks like:
```

Desc    Family ID       human   chimp   orang   baboon  gibbon  macaque marmoset        rat     mouse   cat     horse   cow
(null)  0       66      61      61      62      63      64      66      63      61      59      62      63
(null)  1       52      55      54      57      54      56      56      53      52      57      55      54
(null)  2       76      51      41      39      45      36      37      67      79      37      41      49
```

So your orthogroup field should actually be the second one. (The file you provide here also has DOS line endings, so you will need to run Dos2Unix to get it to run.)

**Reply by MaevaTecher** [reply]
Thank you very much! I completely forgot about adding the description column. It worked now. Just having some issues with large duplication events that make the model unable to converge, but I am sure I will figure it out from others issues posted :)

**Comment by sanvi1804** [comment]
Ok thank you so much
Thanks & Regards,
Garima SaxenaPhD ScholarC/O: Dr. Mehar Hasan AsifComputational BiologyCSIR-National Botanical Research InstituteLucknow, Uttar Pradesh, India 

    On Friday 2 August, 2024 at 12:36:57 am IST, Ben Fulton ***@***.***> wrote:  
 
 


When I run your files I get the error message, "China was not found in gene family OG0000000". In your tree file you have "Nelumbo nucifera China" (with spaces) and in the families you have "Nelumbo_nucifera_China" (with underscores). Try replacing the spaces with underscores in your tree.

**Comment by pedres** [comment]
Same problem here with the file got after orthofinder. I have tried dos2unix and 

dos2unix Orthogroups.GeneCount.txt

### remove column "Total"
awk 'BEGIN{OFS=FS="\t"} {NF--; print}' Orthogroups.GeneCount.txt > Orthogroups.GeneCountOK.txt 
awk 'NR==1 {print "Desc\t" $0; next} {print "A" NR-1 "\t" $0}' Orthogroups.GeneCountOK.txt > Orthogroups.GeneCountOK_with_Desc.txt
## To estimate lambda with no among family rate variation issue the command
cafe5 -c 20 -i Orthogroups.GeneCountOK_with_Desc.txt -t FcC_supermatrix_trimmed.fasROOT_NEWICK.ultrametric.tre -o fixed_gamma

Command line: cafe5 -c 20 -i Orthogroups.GeneCountOK_with_Desc.txt -t FcC_supermatrix_trimmed.fasROOT_NEWICK.ultrametric.tre -o fixed_gamma 
Dattemsi was not found in gene family OG0000000

Thanks a lot from your help


[Orthogroups.GeneCount.txt](https://github.com/user-attachments/files/20421774/Orthogroups.GeneCount.txt)
[FcC_supermatrix_trimmed.fasROOT_NEWICK.ultrametrica.txt](https://github.com/user-attachments/files/20421775/FcC_supermatrix_trimmed.fasROOT_NEWICK.ultrametrica.txt)

**Reply by benfulton** [reply]
Your orthogroups file needs a description field as the first field. If you look at the sample file file it looks like:

```
Desc    Family ID       human   chimp   orang   baboon  gibbon  macaque marmoset        rat     mouse   cat     horse   cow
(null)  0       66      61      61      62      63      64      66      63      61      59      62      63
(null)  1       52      55      54      57      54      56      56      53      52      57      55      54
(null)  2       76      51      41      39      45      36      37      67      79      37      41      49
```

However, your orthogroups file does not seem to have Dattemsi. I see

```
Orthogroup	Acaliginosa	Acorticis	Aicterica	Ailacteoespumosus	Aplanci	Beiseni	Cgigas	Cmatritensis	Eandrei	Efetida	Enchycrypticus	Lgigantea	Lrubellus	Lterrestris	Mvulgaris	Nnajaformis	Total
```

Am I missing something?

**Comment by benlemmond** [comment]
Hi, I am also having this problem (XX sample was not found in gene family), and have tried all the fixes listed above (dos2unix, use .csv, ensure 'Desc' column is first, double-check that all tip labels match column headers *exactly*, taking care to check spaces/etc.) and I still can't find any discrepancies or issues. Can you suggest any fixes? Thank you!

[A_tree_dated_phy.txt](https://github.com/user-attachments/files/23596722/A_tree_dated_phy.txt)
[Orthogroups_GeneCounts_SubcladeA_2.csv](https://github.com/user-attachments/files/23596786/Orthogroups_GeneCounts_SubcladeA_2.csv)

**Reply by benfulton** [reply]
The family file needs to be tab-separated, not comma-separated.
